pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,28 +1,68 @@
 import 'package:flutter/material.dart';
 import 'package:pokedex_asyn_redux/api/model/pokemon.dart';
 import 'package:pokedex_asyn_redux/features/pokemon_overview/widgets/pokemon_card.dart';
+import 'package:pokedex_asyn_redux/utils/async.dart';
+import 'package:pokedex_asyn_redux/utils/color_constants.dart';
 import 'package:pokedex_asyn_redux/utils/constants.dart';
 
 class PokemonOverviewPage extends StatelessWidget {
-  const PokemonOverviewPage({
+  PokemonOverviewPage({
     required this.pokemons,
     Key? key,
   }) : super(key: key);
 
-  final List<Pokemon> pokemons;
+  final Async<List<Pokemon>> pokemons;
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text(pokemonOverviewTitle),
       ),
-      body: GridView.builder(
-        itemCount: pokemons.length,
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
-        itemBuilder: (_, index) {
-          final pokemon = pokemons[index];
-          return PokemonCard(pokemon: pokemon);
+      body: pokemons.when(
+        (data) {
+          return GridView.builder(
+            itemCount: data.length,
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
+            itemBuilder: (_, index) {
+              final pokemon = data[index];
+              return PokemonCard(pokemon: pokemon);
+            },
+          );
+        },
+        loading: () => Container(
+          color: loadingScreen,
+          child: Center(
+            child: Image.asset(
+              pikachuRunningImage,
+              height: 210,
+              width: 210,
+            ),
+          ),
+        ),
+        error: (errorMessage) {
+          WidgetsBinding.instance.addPostFrameCallback((_) {
+            ScaffoldMessenger.of(context).showSnackBar(
+              SnackBar(
+                content: Text(errorMessage!),
+                duration: const Duration(days: 1),
+              ),
+            );
+          });
+          return Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                const Text(noPokemonsAvailableLabel),
+                const SizedBox(height: 20),
```

Comment:
maybe you can create a reusable widgets for height and width spacings instead of doing manually every time 🙂

name those widgets as:
- VerticalSpace
- HorizontalSpace